waspc/e2e-test/Util.hs

```diff
@@ -0,0 +1,82 @@
+module Util
+  ( newCommand,
+    newAndCompileCommand,
+    newAndCompileAndMigrateCommand,
+    runGoldenTest,
+  )
+where
+
+import Control.Monad (filterM)
+import Data.List (intercalate, sort)
+import Data.Text (pack, replace, unpack)
+import System.Directory (doesFileExist)
+import System.Directory.Recursive (getDirFiltered)
+import System.FilePath (joinPath, makeRelative, takeFileName, (</>))
+import System.Process (callCommand)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Golden (goldenVsFileDiff)
+
+-- TODO: Switch over from FilePath to StrongPath eventually
+
+-- TODO: Handle shell command descriptions/invocations and file manipulation in some better way (and more platform agnostic?)
+-- NOTE: Using `wasp-cli` herein so we can assume using latest `stack install` in CI and locally. Better way?
+newCommand :: String -> String
+newCommand projectName =
+  intercalate " && " ["wasp-cli new " ++ projectName, "cp " ++ ("overlays" </> ".gitignore") ++ " " ++ projectName]
+
+newAndCompileCommand :: String -> String
+newAndCompileCommand projectName =
+  intercalate " && " [newCommand projectName, "cd " ++ projectName, "wasp-cli compile"]
+
+newAndCompileAndMigrateCommand :: FilePath -> FilePath -> String -> String
+newAndCompileAndMigrateCommand _cwd overlayFp projectName =
+  let generatedMigrationsDir = joinPath [".wasp", "out", "db", "migrations"]
+      waspMigrationsDir = "migrations"
+      migrationName = "foo"
+   in intercalate
+        " && "
+        [ newAndCompileCommand projectName,
+          "cat " ++ overlayFp ++ " >> main.wasp",
+          "wasp-cli db migrate-dev " ++ migrationName,
+          "mv " ++ (waspMigrationsDir </> ("*" ++ migrationName)) ++ " " ++ (waspMigrationsDir </> ("no-date-" ++ migrationName)),
+          "mv " ++ (generatedMigrationsDir </> ("*" ++ migrationName)) ++ " " ++ (generatedMigrationsDir </> ("no-date-" ++ migrationName))
+        ]
+
+runGoldenTest :: FilePath -> String -> (String -> String) -> IO TestTree
+runGoldenTest cwd goldenDirName command = do
+  let testDirName = goldenDirName ++ "Test"
+  let command' = command testDirName
```

Comment:
this is later used only once, so maybe not worth extracting it?

Reply:
Good point, I could have also renamed the identifier to be `commandRunningRelativeToTestOutputDir` lol ok maybe not, but that is what it does when you give it the dir, constructs all commands relative to that FP. I'll just not extract for now.

Reply:
makes sense! although I like your long name also :D